### PR TITLE
Re-enabled Azure - requested in bagetter/BaGetter#28

### PR DESCRIPTION
**Pull request by seriouz**
As request in https://github.com/bagetter/BaGetter/issues/28 i re-enabled the commented out Azure part.
Sadly i couldn't test it because i have no Azure access at the moment.

### CODE REVIEW

File: src/BaGetter.Azure/Storage/BlobStorageService.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public async Task<Stream> GetAsync(string path, CancellationToken cancellationTo
         {
             return await _container
                 .GetBlockBlobReference(path)
-                .OpenReadAsync(cancellationToken);
+                .OpenReadAsync();
```
Comment (Regenhardt) [inline src/BaGetter.Azure/Storage/BlobStorageService.cs:25]: Why not pass the token?

File: src/BaGetter.Azure/Storage/BlobStorageService.cs [hunk 2]
```diff
@@ -65,7 +65,7 @@ public Task<Uri> GetDownloadUriAsync(string path, CancellationToken cancellation
             }
             catch (StorageException e) when (e.IsAlreadyExistsException())
             {
-                using (var targetStream = await blob.OpenReadAsync(cancellationToken))
+                using (var targetStream = await blob.OpenReadAsync())
```
Comment (Regenhardt) [inline src/BaGetter.Azure/Storage/BlobStorageService.cs:68]: No token either?

File: src/BaGetter.Azure/Storage/BlobStorageService.cs [hunk 3]
```diff
@@ -79,7 +79,7 @@ public async Task DeleteAsync(string path, CancellationToken cancellationToken)
         {
             await _container
                 .GetBlockBlobReference(path)
-                .DeleteIfExistsAsync(cancellationToken);
+                .DeleteIfExistsAsync();
```
Comment (Regenhardt) [inline src/BaGetter.Azure/Storage/BlobStorageService.cs:82]: This starts to look planned, anything about cancellation tokens I should know?

**Review by Regenhardt — COMMENTED** [review]
Will try to get it running on Azure to test it. Can't promise anything though, so I'd be glad if anyone else jumps in for me here.

File: src/BaGetter.Azure/Storage/BlobStorageService.cs (hunk 1 above)
Comment (seriouz) [reply src/BaGetter.Azure/Storage/BlobStorageService.cs:25]: The api changed: They discarded the CT as single argument /facepalm

File: src/BaGetter.Azure/Storage/BlobStorageService.cs (hunk 2 above)
Comment (seriouz) [reply src/BaGetter.Azure/Storage/BlobStorageService.cs:68]: The api changed: They discarded the CT as single argument /facepalm

File: src/BaGetter.Azure/Storage/BlobStorageService.cs (hunk 3 above)
Comment (seriouz) [reply src/BaGetter.Azure/Storage/BlobStorageService.cs:82]: The api changed: They discarded the CT as single argument /facepalm

**Comment by Regenhardt** [comment]
Tried to run it locally connecting to Azure Storage + Search, got the following message ("package-search" is my search instance name):

```
Microsoft.Rest.Azure.CloudException: The index 'packages' for service 'package-search' was not found.

   at Microsoft.Azure.Search.DocumentsProxyOperations.ContinueSearchPostWithHttpMessagesAsync[T](String _url, SearchRequest searchRequest, Nullable`1 clientRequestId, Dictionary`2 customHeaders, Boolean _shouldTrace, String _invocationId, CancellationToken cancellationToken, JsonSerializerSettings requestSerializerSettings, JsonSerializerSettings responseDeserializerSettings)

   at Microsoft.Azure.Search.DocumentsProxyOperations.SearchPostWithHttpMessagesAsync[T](SearchRequest searchRequest, SearchRequestOptions searchRequestOptions, Dictionary`2 customHeaders, CancellationToken cancellationToken, JsonSerializerSettings requestSerializerSettings, JsonSerializerSettings responseDeserializerSettings)

   at Microsoft.Azure.Search.DocumentsOperationsExtensions.SearchAsync[T](IDocumentsOperations operations, String searchText, SearchParameters searchParameters, SearchRequestOptions searchRequestOptions, CancellationToken cancellationToken)

   at BaGetter.Azure.AzureSearchService.SearchAsync(SearchRequest request, CancellationToken cancellationToken) in E:\Repo\BaGetter\src\BaGetter.Azure\Search\AzureSearchService.cs:line 48

   at BaGetter.Web.IndexModel.OnGetAsync(CancellationToken cancellationToken) in E:\Repo\BaGetter\src\BaGetter.Web\Pages\Index.cshtml.cs:line 49

   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.ExecutorFactory.GenericTaskHandlerMethod.Convert[T](Object taskAsObject)

   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.ExecutorFactory.GenericTaskHandlerMethod.Execute(Object receiver, Object[] arguments)

   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeHandlerMethodAsync()

   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeNextPageFilterAsync()

   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.Rethrow(PageHandlerExecutedContext context)

   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeInnerFilterAsync()

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)

   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|7_0(Endpoint endpoint, Task requestTask, ILogger logger)

   at BaGetter.Web.OperationCancelledMiddleware.Invoke(HttpContext context) in E:\Repo\BaGetter\src\BaGetter.Web\OperationCancelledMiddleware.cs:line 28

   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)

   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
```

It it possible to automatically add this index, or should we rather add this as a config step to the installation instructions?

**Comment by Regenhardt** [comment]
Also I couldn't get the existing Dockerfile working in a azure web app, still working on it since I'm not sure what would the correct way be to both use docker and configure azure services.

Tried deleting env settings from the Dockerfile and configuring azure services in the appsettings.json, didn't work. This was quick and dirty anyway.   
I guess I'm gonna try to put the configuration stuff into the azure web app configuration and test that, but definitely tomorrow because it's late already and I have to sleep.

In case anyone has a moment while I sleep (and work a bit tomorrow), this is the output of the azure web app log stream on my latest attempt, maybe someone knows this and has a quick answer to what's wrong:

```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1"/><title>502 - Web server received an invalid response while acting as a gateway or proxy server.</title><style type="text/css"><!--body{margin:0;font-size:.7em;font-family:Verdana, Arial, Helvetica, sans-serif;background:#EEEEEE;}fieldset{padding:0 15px 10px 15px;}h1{font-size:2.4em;margin:0;color:#FFF;}h2{font-size:1.7em;margin:0;color:#CC0000;}h3{font-size:1.2em;margin:10px 0 0 0;color:#000000;}#header{width:96%;margin:0 0 0 0;padding:6px 2% 6px 2%;font-family:"trebuchet MS", Verdana, sans-serif;color:#FFF;background-color:#555555;}#content{margin:0 0 0 2%;position:relative;}.content-container{background:#FFF;width:96%;margin-top:8px;padding:10px;position:relative;}--></style></head><body><div id="header"><h1>Server Error</h1></div><div id="content"><div class="content-container"><fieldset><h2>502 - Web server received an invalid response while acting as a gateway or proxy server.</h2><h3>There is a problem with the page you are looking for, and it cannot be displayed. When the Web server (while acting as a gateway or proxy) contacted the upstream content server, it received an invalid response from the content server.</h3></fieldset></div></div></body></html>
```

**Comment by Regenhardt** [comment]
Ok deploy to Azure app service is configured correctly, using the actual config settings of Azure is both a good idea and actually works. However, there is still the problem with the non-existing index (see above exception).

**Comment by Regenhardt** [comment]
I'd prefer the search provider to create the index if it doesn't exist, as that would make it easier to make an Azure template that just creates the required resources and the BaGetter container does the config on startup.

**Comment by seriouz** [comment]
This is a good idea!
Can you look into this? Currently I can only look during my evenings into this.

**Comment by yaegashi** [comment]
Hi, thank you all for working on this issue!

@Regenhardt Unfortunately, I have no experience with Azure AI Search...

There was BaGet.Tools.AzureSearchImporter in BaGet, but it seems to have been removed.
- https://github.com/loic-sharma/BaGet/issues/20
- https://github.com/loic-sharma/BaGet/commit/3a2ba5e2eb9e084666c1c3b721a83f671e81ab42

I guess adding some packages with `dotnet nuget push` (before opening the website with a web browser) may create a new index "packages".

**Comment by seriouz** [comment]
Can some provide an Azure Search API key for me, maybe then i'll could help.
You can find me [here in Discord](https://discord.com/invite/XsAmm6f2hZ).

**Comment by Regenhardt** [comment]
Once we get this updated to the new APIs, we could also add a template to create all the required Azure services in one go.

**Comment by seriouz** [comment]
@daghsentinel Do you have a look into this Azure stuff currently? I started with the new "Azure.*"-libraries locally, but got stuck on Azure Cosmos. If you have interest in my progress give me a nugde ^^

File: src/BaGetter.Azure/BaGetter.Azure.csproj [hunk 4]
```diff
@@ -11,14 +11,16 @@
     <PackageReference Include="Azure.Search.Documents" />
     <PackageReference Include="Azure.Storage.Blobs" />
     <PackageReference Include="Microsoft.Azure.Cosmos" />
+    <PackageReference Include="Microsoft.Azure.Cosmos.Table" />
+    <PackageReference Include="Microsoft.Azure.Search" />
+    <PackageReference Include="Microsoft.Azure.Search.Data" />
+    <PackageReference Include="Microsoft.Azure.Search.Service" />
     <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" />
+    <PackageReference Include="WindowsAzure.Storage" />
```
Comment (Regenhardt) [inline src/BaGetter.Azure/BaGetter.Azure.csproj:19]: `WindowsAzure.Storage` is deprecated, NuGet gallery is just slow at upgrading it.

**Review by Regenhardt — COMMENTED** [review]
Don't worry about fixing this, I just use it as reference.

**Pull request closed without merging**